### PR TITLE
[JavaScript] Fixed the filter does not work in the transaction table of Explorer Webchaind. No. 0001

### PR DESCRIPTION
**Pull request by laitaiyu**
Fixed the filter does not work in the transaction table of Explorer Webchaind. No. 0001

### CODE REVIEW

File: public/js/controllers/AddressController.js [hunk 1]
```diff
@@ -42,12 +42,73 @@ angular.module('BlocksApp').controller('AddressController', function($stateParam
           data.addr = $scope.addrHash;
           data.count = $scope.addr.count;
           $http.post('/addr', data).then(function(resp) {
+          $('label input').on('keyup', function () {
+              resp.data.keyword = $('label input').val();
+          });
+          var search = $('label input').val() ? $('label input').val() : null;
```
Comment (fedorenko22116) [inline public/js/controllers/AddressController.js]: Variables with jquery selectors should be prefixed with `$`

File: public/js/controllers/AddressController.js [hunk 2]
```diff
@@ -42,12 +42,73 @@ angular.module('BlocksApp').controller('AddressController', function($stateParam
...
+              data: [[],[],[],[],[],[],[],[],[],[]]
+          };
+          clone.draw = resp.data.draw;
+          clone.recordsFiltered = resp.data.recordsFiltered;
+          clone.recordsTotal = resp.data.recordsTotal;
+          clone.mined = resp.data.mined;
+          clone.data = resp.data.data;
+          var clearItem = 0;
+          if (search != null) {
+              for (var iI = 0; iI < clone.data.length; iI++) {
```
Comment (fedorenko22116) [inline public/js/controllers/AddressController.js]: Use human-readable names or even replace it with `forEach`

File: public/js/controllers/AddressController.js [hunk 3]
```diff
@@ -42,12 +42,73 @@ angular.module('BlocksApp').controller('AddressController', function($stateParam
...
+                  var arrayItem = [];
+                  var vDeleteBlockAndDate = 0;
+                  arrayItem = clone.data[iI];
+                  for (var iJ = 0; iJ < 7; iJ++) {
```
Comment (fedorenko22116) [inline public/js/controllers/AddressController.js]: The same as in previous comment

File: public/js/controllers/AddressController.js [hunk 4]
```diff
@@ -42,12 +42,73 @@ angular.module('BlocksApp').controller('AddressController', function($stateParam
...
+                      if (iJ == 1) {
```
Comment (fedorenko22116) [inline public/js/controllers/AddressController.js]: Why don't you use `switch` operator to compare same value

File: public/js/controllers/AddressController.js [hunk 5]
```diff
@@ -42,12 +42,73 @@ angular.module('BlocksApp').controller('AddressController', function($stateParam
...
+                          }
+                      }
+                  }
+                  if (vDeleteBlockAndDate == 2) {
+                      clone.data.splice(iI,1);
+                      iI = -1;
+                  }
+              }
+          }
+          if (clearItem == 0 && search != null) {
```
Comment (fedorenko22116) [inline public/js/controllers/AddressController.js]: use equal type operators to compare objects

File: public/js/controllers/AddressController.js (hunk 3 above)
Comment (fedorenko22116) [inline public/js/controllers/AddressController.js]: hardcoded (aka magic var) length

File: public/js/controllers/AddressController.js [hunk 6]
```diff
@@ -42,12 +42,73 @@ angular.module('BlocksApp').controller('AddressController', function($stateParam
           data.addr = $scope.addrHash;
           data.count = $scope.addr.count;
           $http.post('/addr', data).then(function(resp) {
+          $('label input').on('keyup', function () {
+              resp.data.keyword = $('label input').val();
```
Comment (fedorenko22116) [inline public/js/controllers/AddressController.js]: `$('label').find('input').val();`

File: public/js/controllers/AddressController.js (hunk 1 above)
Comment (fedorenko22116) [reply public/js/controllers/AddressController.js]: can be simplified to `$('label').find('input').val() || null`

**Review by fedorenko22116 — CHANGES_REQUESTED** [review]

File: public/js/controllers/AddressController.js (hunk 5 above)
Comment (laitaiyu) [reply public/js/controllers/AddressController.js]: I use the strict equality operators to compare variables.

**Comment by laitaiyu** [comment]
I fixed whole variables for human-readable.
I agreed the reviewer advice and I fixed those code.
The code had tested by myself and it's work well.

**Review by fedorenko22116 — APPROVED** [review]

**Pull request merged**